pulpcore/cli/common.py

```diff
@@ -20,20 +19,21 @@ def output_result(self, result):
         if self.format == "json":
             click.echo(json.dumps(result, cls=PulpJSONEncoder, indent=2))
 
-    def call(self, operation_id, *args, **kwargs):
+    def call(self, operation_id, background=False, *args, **kwargs):
         result = self.api.call(operation_id, *args, **kwargs)
         if "task" in result:
             task_href = result["task"]
             click.echo(f"Started task {task_href}", err=True)
-            result = self.wait_for_task(task_href)
-            click.echo("Done.", err=True)
+            if not background:
+                result = self.wait_for_task(task_href)
+                click.echo("Done.", err=True)
         return result
 
     def wait_for_task(self, task_href, timeout=120):
         while timeout:
             timeout -= 1
             task = self.api.call("tasks_read", parameters={"task_href": task_href})
-            if task["state"] == "completed":
+            if task["state"] == "completed" or task["state"] == "canceled":
```

Comment:
I am leaning to throwing an Exception for state "cancel" to properly indicate in the "normal" commands that your task didn't finish successfully.
You'd need to ignore this exception in the cancel command obviously.